tools/tf2openapi/types/tftensor.go

```diff
@@ -34,30 +37,80 @@ type TFDType int
 const (
 	// all the possible constants that can be JSON-ified according to
 	// https://www.tensorflow.org/tfx/serving/api_rest#json_mapping
-	DT_BOOL TFDType = iota
-	DT_B64_STRING
-	DT_STRING
-	DT_INT8
-	DT_UINT8
-	DT_INT16
-	DT_INT32
-	DT_UINT32
-	DT_INT64
-	DT_UINT64
-	DT_FLOAT
-	DT_DOUBLE
+	// along with a representation for B64 strings
+	DtBool TFDType = iota
+	DtB64String
+	DtString
+	DtInt8
+	DtUInt8
+	DtInt16
+	DtInt32
+	DtUInt32
+	DtInt64
+	DtUint64
+	DtFloat
+	DtDouble
 )
 
-func NewTFTensor(key string, tensorInfo pb.TensorInfo) TFTensor {
-	return TFTensor{}
+func NewTFTensor(key string, tensor pb.TensorInfo) TFTensor {
+	if tensor.TensorShape.UnknownRank {
+		return TFTensor{
+			Key:   key,
+			DType: NewTFDType(tensor.Dtype.String(), key),
+			Shape: nil,
+			Rank:  -1,
+		}
+	} else {
+		tfShape := NewTFShape(tensor.TensorShape.Dim)
+		return TFTensor{
+			Key:   key,
+			DType: NewTFDType(tensor.Dtype.String(), key),
+			Shape: tfShape,
+			Rank:  int64(len(tfShape)),
+		}
+	}
 }
 
 func NewTFShape(dimensions []*fw.TensorShapeProto_Dim) TFShape {
-	return TFShape{}
+	tfShape := TFShape{}
+	// There will always be -1 in dimensions[0] for batch size, so ignore
+	for _, d := range dimensions[1:] {
+		tfShape = append(tfShape, d.Size)
+	}
+	return tfShape
 }
 
 func NewTFDType(dType string, name string) TFDType {
-	return TFDType(0)
+	switch dType {
```

Comment:
Would it be better to define this as a map? Might be a little challenging with the if else piece but you could separate that.

Reply:
I think a map would be better because it's more readable.